src/Accordion.js

```diff
@@ -0,0 +1,53 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { mapToCssModules, tagPropType } from './utils';
+import { AccordionContext } from './AccordionContext';
+
+const propTypes = {
+  tag: tagPropType,
+  className: PropTypes.string,
+  cssModule: PropTypes.object,
+  innerRef: PropTypes.oneOfType([
+    PropTypes.object,
+    PropTypes.string,
+    PropTypes.func,
+  ]),
+  children: PropTypes.node,
+};
+
+const defaultProps = {
+  tag: 'div'
+};
+
+const Accordion = (props) => {
+  const {
+    className,
+    cssModule,
+    tag: Tag,
+    innerRef,
+    ...attributes
+  } = props;
+  const classes = mapToCssModules(classNames(
+    className,
+    'accordion',
+  ), cssModule);
+
+  const [openId, setOpenId] = useState();
+
+  return (
+    <AccordionContext.Provider value={{
```

Comment:
Let's `useMemo` the value for this context

Reply:
Curious why we should do this - the value doesn't really require expensive computation to come up with 🤔 I'd think that adding a useMemo hook here would make things more complex without much benefit.

Reply:
It's to avoid re-rendering of the context consumers in the case the accordion needs to re-render and the context object values don't change.  Without `useMemo`, the object reference changes each time, so the `AccordionContext.Consumer`s will re-render unnecessarily.

Reply:
Ahh sorry - was assuming we were going to convert to `useContext` per the other comment. Re-rendering shouldn't be a concern if we convert to the hook, right?

Reply:
We would use a combination of `useMemo` (on the provider side) and `useContext` (on the consumer side).  `useContext` mainly because you can avoid the `Context.Consumer` declaration and in general it's a much cleaner look :)

For example you can reduce something like
```
return (
  <AccordionContext.Consumer>
    {{ openId } => {
      return ... 
    }
  <AccordionContext.Consumer>
);
```

to

```
const { openId } = useContext(AccordionContext);

return ...
```